pandas/tseries/holiday.py

```diff
@@ -354,7 +354,9 @@ def _apply_rule(self, dates: DatetimeIndex) -> DatetimeIndex:
         Dates with rules applied
         """
         if dates.empty:
-            return DatetimeIndex([])
+            # Handle case where we get an empty series of dates with a
+            # defined timezone
+            return DatetimeIndex([]).tz_localize(dates.tz)
```

Comment:
```suggestion
            return dates.copy()
```

Reply:
Didn't realize that was a thing -- nifty.